### PR TITLE
Update documentation for parameter shape placeholder in MNIST example.

### PR DESCRIPTION
**Pull request by zori**
Documentation to https://github.com/pfnet/chainer/pull/1551

### CODE REVIEW

File: chainer/training/updater.py [hunk 1]
```diff
@@ -289,8 +289,8 @@ def update_core(self):
             in_arrays_list[key] = self.converter(
                 batch[i::n], self._devices[key])
 
-        for model in six.itervalues(self._models):
-            model.cleargrads()
+        # for model in six.itervalues(self._models):
```
Comment (okuta) [inline chainer/training/updater.py]: Why do you comment out this line?

File: docs/source/tutorial/basic.rst [hunk 2]
```diff
@@ -556,5 +557,8 @@ These extensions perform the following tasks:
 There are many extensions implemented in the :mod:`chainer.training.extensions` module.
 The most important one that is not included above is :func:`~training.extensions.snapshot`, which saves the snapshot of the training procedure (i.e., the Trainer object) to a file in the output directory.
 
-The example code in the `examples/mnist` directory contains GPU support, though the essential part is same as the code in this tutorial.
-We will review in later sections how to use GPU(s).
+The `example code <https://github
```
Comment (okuta) [inline docs/source/tutorial/basic.rst]: Is line break required?

File: docs/source/tutorial/gpu.rst [hunk 3]
```diff
@@ -153,27 +153,6 @@ In this subsection, the code is based on :ref:`our first MNIST example in this t
 
 A :class:`Link` object can be transferred to the specified GPU using the :meth:`~Link.to_gpu` method.
 
-.. testcode::
```
Comment (okuta) [inline docs/source/tutorial/gpu.rst]: Do not remove this test code.
Please fix this.

You can use command to test for documents.
```bash
cd docs
make doctest
```

File: chainer/training/updater.py (hunk 1 above)
Comment (zori) [inline chainer/training/updater.py]: My mistake, this should not have been committed here. I wrote my remark here: https://github.com/pfnet/chainer/pull/1551#discussion_r77295028

Comment (zori) [inline chainer/training/updater.py]: I `--amend`-ed the previous commit to exclude updater.py, as it was not my intention to include it in the PR.

File: docs/source/tutorial/basic.rst (hunk 2 above)
Comment (zori) [inline docs/source/tutorial/basic.rst]: fixed, thank you

File: docs/source/tutorial/gpu.rst (hunk 3 above)
Comment (zori) [inline docs/source/tutorial/gpu.rst]: Sorry, I had misunderstood this section. Fixed

**Comment by okuta** [comment]
Thank you!

**Pull request merged**